Source/PINDiskCache.m

```diff
@@ -1397,20 +1439,42 @@ - (void)setWritingProtectionOption:(NSDataWritingOptions)writingProtectionOption
     NSDataWritingOptions option = NSDataWritingFileProtectionMask & writingProtectionOption;
     
     [strongSelf lock];
-    strongSelf->_writingProtectionOption = option;
+        strongSelf->_writingProtectionOption = option;
     [strongSelf unlock];
   } withPriority:PINOperationQueuePriorityHigh];
 }
 #endif
 
+- (void)lockUntilWritable
+{
+    [self lock];
+    
+    // spinlock if the disk isn't writable
+    if (_diskWritable == NO) {
+        pthread_cond_wait(&_diskWritableCondition, &_mutex);
+    }
+}
+
+- (void)lockUntilDiskStateKnown
+{
+    [self lock];
+    
+    // spinlock if the disk state isn't known
+    if (_diskStateKnown == NO) {
+        pthread_cond_wait(&_diskStateKnownCondition, &_mutex);
+    }
+}
```

Comment:
Two alternate names might be `lockForWriting` and `lockAndWaitForKnownDiskState` or something. `lockUntil` isn't quite on the money.

Reply:
I like those better.